pkg/processor/trigger/partitioned/v3io/partition.go

```diff
@@ -131,3 +136,29 @@ func (p *partition) getRecords(partitionPath string, location string, pollingInt
 	return response.Output.(*v3iohttp.GetRecordsOutput), nil
 
 }
+
+func (p *partition) waitPartitionAvailable(partitionPath string, pollingInterval time.Duration) error {
+	responseChannel := make(chan *v3iohttp.Response)
+
+	for {
+		request := &v3iohttp.ListBucketInput{
+			Path: p.v3ioTrigger.streamPath,
+		}
+		p.v3ioTrigger.container.ListBucket(request, context.TODO(), responseChannel)
+		response := <-responseChannel
+
+		listBucketResult, ok := response.Output.(*v3iohttp.ListBucketOutput)
+		if !ok {
+			p.Logger.ErrorWith("Failed to list bucket", "err", response.Error)
+			time.Sleep(pollingInterval)
+			continue
+		}
+
+		for _, partition := range listBucketResult.Contents {
+			if partition.Key == partitionPath {
```

Comment:
I'm worried that this might fail on something like an extra slash (i.e. the comparison is not canonical).

Reply:
`partitionPath` is being constructed by trigger, `partition.Key` has a strict structure. no need for defensive programming here

Reply:
But `partitionPath` is constructed from `p.v3ioTrigger.streamPath`, which is derived from a user-defined property 😅. I'm pretty sure a trigger URL containing extra slashes will mess this up (based on playing around with `v3ioUtilTestSuite`).